mmcls/evaluation/metrics/multi_task.py

```diff
@@ -2,24 +2,85 @@
 from typing import Dict, List, Sequence
 
 from mmengine.evaluator import BaseMetric
-from mmengine.registry import METRICS as mmengineMetrics
 
 from mmcls.registry import METRICS
+from mmcls.structures import MultiTaskDataSample
 
 
 @METRICS.register_module()
-class MultiTasks(BaseMetric):
-    """a."""
+class MultiTasksMetric(BaseMetric):
+    """Metrics for MultiTask
+    Args:
+        task_metrics(dict): a dictionary in the keys are the names of the tasks
+            and the values is a list of the metric corresponds to this task
+    Examples:
+        task_metrics = {
+            'task0': [dict(type='Accuracy', topk=(1, ))],
+            'task1': [
+                dict(type='Accuracy', topk=(1, 3)),
+                dict(type='SingleLabelMetric', items=['precision', 'recall'])
+            ]
+        }
+        task_metrics2 = {
+            'task0': [dict(type='Accuracy', topk=(1, ))],
+            'task1': {
+                "task10": [
+                    dict(type='Accuracy', topk=(1, 3)),
+                    dict(type='SingleLabelMetric', items=['precision'])
+                ],
+                "task11": [dict(type='Accuracy', topk=(1, ))]
+            }
+        }
+    """
 
     def __init__(self,
                  task_metrics: Dict,
                  collect_device: str = 'cpu') -> None:
         self.task_metrics = task_metrics
         super().__init__(collect_device=collect_device)
 
+        self._metrics = {}
         for task_name in self.task_metrics.keys():
-            globals()['metric_%s' % task_name] = mmengineMetrics.build(
-                self.task_metrics[task_name])
+            if type(self.task_metrics[task_name]) == list:
+                self._metrics[task_name] = []
+                for metric in self.task_metrics[task_name]:
+                    self._metrics[task_name].append(METRICS.build(metric))
+            elif type(self.task_metrics[task_name]) == dict:
```

Comment:
you need to implement a recursive strategy here, it will fail if you have a 3-level nested tasks